### PR TITLE
Detect copy replacing that alter replace

### PR DESCRIPTION
**Pull request by smedilol**
Add detection of replace altered by a replacing clause.

### CODE REVIEW

**Comment by smedilol** [comment]
Another TODO, `ProcessingDiagnostics` must be available outside `#if EUROINFO_RULES`.

File: TypeCobol/Compiler/Preprocessor/ReplacingTokensLinesIterator.cs [hunk 1]
```diff
@@ -22,12 +24,43 @@ public ReplacingTokensLinesIterator(ITokensLinesIterator sourceIterator, CopyDir
...
 
-        protected override CheckTokenStatus CheckNextTokenBeforeReplace(IReadOnlyList<ReplaceOperation> currentReplaceOperations)
+        protected override CheckTokenStatus CheckNextTokenBeforeReplace([CanBeNull] IReadOnlyList<ReplaceOperation> currentReplaceOperations)
         {
             var nextToken = SourceIteratorNextToken();
 
-            // TODO: Add error to signal unsupported feature or implement
-            // This parser currently does not support REPLACE coming from a copy and that should be altered by the REPLACING clause
+            // This parser currently does not support REPLACE that are altered by the REPLACING clause
+            if (nextToken.TokenType == TokenType.REPLACE_DIRECTIVE)
```
Comment (fm-117) [inline TypeCobol/Compiler/Preprocessor/ReplacingTokensLinesIterator.cs:32]: replace with `while` loop like in ReplaceIterator ?

File: TypeCobol/Compiler/Preprocessor/ReplacingTokensLinesIterator.cs [hunk 2]
```diff
@@ -22,12 +24,43 @@ public ReplacingTokensLinesIterator(ITokensLinesIterator sourceIterator, CopyDir
...
+            {
+                //TODO ReplaceAndReplacing better code to find the real ReplaceDirective on the line. Property ReplaceDirective is the LAST on the line
```
Comment (fm-117) [inline TypeCobol/Compiler/Preprocessor/ReplacingTokensLinesIterator.cs:34]: The ReplaceIterator casts the token into `CompilerDirectiveToken` to retrieve the directive but your code should work too.

File: TypeCobol/Compiler/Scanner/TokensIterator.cs [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        // Source data
+        private string textName;
+        private IList<Token> tokens;
+
+        private int _tokenIndexInLine;
+        
+
+        // Current token
+        private Token currentToken;
+        private int _tokenIndexInLineSnapshot;
```
Comment (fm-117) [inline TypeCobol/Compiler/Scanner/TokensIterator.cs:21]: This is more 'index in list' right ?

**Review by fm-117 — CHANGES_REQUESTED** [review]

**Review by fm-117 — APPROVED** [review]

**Pull request merged**